File: Server/src/main/java/com/api/server/foodb/CompoundsFlavor.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "compounds_flavors")
public class CompoundsFlavor {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @NotNull
    @Column(name = "compound_id", nullable = false)
    private Integer compoundId;

    @NotNull
    @Column(name = "flavor_id", nullable = false)
    private Integer flavorId;

    @NotNull
    @Lob
    @Column(name = "citations", nullable = false)
    private String citations;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @Column(name = "creator_id")
    private Integer creatorId;

    @Column(name = "updater_id")
    private Integer updaterId;

    @Column(name = "source_id")
    private Integer sourceId;

    @Size(max = 255)
    @Column(name = "source_type")
    private String sourceType;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/CompoundsFlavor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("compounds_flavors")]
[Index("CompoundId", "FlavorId", Name = "index_compounds_flavors_on_compound_id_and_flavor_id", IsUnique = true)]
[Index("SourceId", "SourceType", Name = "index_compounds_flavors_on_source_id_and_source_type")]
public class CompoundsFlavor
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("compound_id")] public int CompoundId { get; set; }

    [Column("flavor_id")] public int FlavorId { get; set; }

    [Column("citations", TypeName = "mediumtext")]
    public string Citations { get; set; } = null!;

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }

    [Column("creator_id")] public int? CreatorId { get; set; }

    [Column("updater_id")] public int? UpdaterId { get; set; }

    [Column("source_id")] public int? SourceId { get; set; }

    [Column("source_type")] public string? SourceType { get; set; }
}